src/ReadLog.jl

```diff
@@ -0,0 +1,134 @@
+# Read LCM log files directly
+
+# export
+#   LCMlog,
+#   close,
+#   readNextEvent,
+#   handle,
+#   subscribe
+
+# also see JuliaLang's RawFD type, but little information found.
+# www.c4learn/c-programming/c-file-structure-and-file-pointer
+mutable struct FILE
+  level::Cshort
+  token::Cshort
+  bsize::Cshort
+  fd::Cuchar
+  flags::Cuint  # (unsigned flags ;)
+  hold::Cuchar
+  buffer::Ptr{Cuchar}
+  curp::Ptr{Cuchar}
+  istemp::Cuint # (unsigned is temp)
+end
+
+mutable struct lcm_eventlog_t
+  f::Ptr{FILE}
+  eventcount::Clonglong
+end
+const _LCMlog = Ptr{lcm_eventlog_t}
+
+
+struct lcm_eventlog_event_t
+  eventnum::Clonglong
+  timestamp::Clonglong
+  channellen::Cint
+  datalen::Cint
+  channel::Ptr{Cuchar}
+  data::Ptr{Cuchar}
+end
+
+
+function lcm_eventlog_create(file::S where S <: AbstractString)
+  ccall(
+    (:lcm_eventlog_create, LCMCore.liblcm),
+    _LCMlog, #Ptr{lcm_eventlog_t},
+    (Cstring, Cstring),
+    file, "r"
+  )
+end
+
+
+function lcm_eventlog_destroy(_log::Ptr{lcm_eventlog_t})
+  ccall(
+    (:lcm_eventlog_destroy, LCMCore.liblcm),
+    Void,
+    (Ptr{lcm_eventlog_t}, ),
+    _log
+  )
+end
+
+
+function lcm_eventlog_read_next_event(_log::Ptr{lcm_eventlog_t})
+    ccall(
+      (:lcm_eventlog_read_next_event, LCMCore.liblcm),
+      Ptr{lcm_eventlog_event_t},
+      (Ptr{lcm_eventlog_t},),
+      _log
+    )
+end
+
+#
+# Doesn't work yet
+# event = unsafe_wrap(lcm_eventlog_event_t, _event, dim??)
+
+
+
+mutable struct LCMlog
+  _log::_LCMlog
+  subscriptions::Dict{AbstractString, LCMCore.SubscriptionOptions}
+  function LCMlog(filename::S) where {S <: AbstractString}
+    try
```

Comment:
This `try-catch` also seems unnecessary. Opening a nonexistent logfile doesn't actually throw, it just returns a log with its `_log` field set to NULL.

I think that we should remove this `try-catch` block and modify `lcm_eventlog_create` to throw an error if its `ccall` returns NULL.

Reply:
Can do that.

Reply:
Thank you!